heron/api/src/java/com/twitter/heron/streamlet/Config.java

```diff
@@ -109,19 +148,24 @@ public Builder setUserConfig(String key, Object value) {
     }
 
     /**
-     * Sets the topology to use the Kryo serializer for serializing
+     * Sets the topology to use the specified serializer for serializing
      * streamlet elements
+     * @param topologySerializer The serializer to be used in this topology
      */
-    public Builder useKryoSerializer() {
-      try {
-        config.setSerializationClassName(new KryoSerializer().getClass().getName());
-      } catch (NoClassDefFoundError e) {
-        throw new RuntimeException("Linking with kryo is needed because useKryoSerializer is used");
+    public Builder setSerializer(Serializer topologySerializer) {
+      serializer = topologySerializer;
```

Comment:
Suggest to have two stages.

setSerializer() function set the variable. And have another applySerializer() to be called in build() function.

Reply:
Good idea. Done.